packages/eas-cli/src/commands/channel/rollout.ts

```diff
@@ -0,0 +1,295 @@
+    type: 'text',
+    name: 'name',
+    format: value => {
+      return parseInt(value, 10);
+    },
+    message: promptMessage,
+    initial: 0,
+    validate: (percent: string): true | string => {
+      /**
+       * Don't allow 100 to emphasise to the developer that rollouts have an open right hand endpoint: [0,N)
+       *
+       * This is unavoidable since the comparision must either be a strict inequality or a weak one. A weak
+       * comparison would result in (0,100]. Assuming the new branch is more dangerous, we choose [0,N) to allow
+       * '0 percent' rollouts.
+       */
+      const floatValue = parseFloat(percent);
+      return Number.isInteger(floatValue) && floatValue >= 0 && floatValue <= 99
+        ? true
+        : validationMessage;
+    },
+  });
+  return percent;
+}
+
+export default class ChannelRollout extends Command {
+  static hidden = true;
+  static description = 'Rollout a new branch out to a channel incrementally.';
+
+  static args = [
+    {
+      name: 'channel',
+      required: true,
+      description: 'rollout that the channel is on',
+    },
+  ];
+
+  static flags = {
+    branch: flags.string({
+      description: 'branch to rollout',
+      required: false,
+    }),
+    percent: flags.integer({
+      description: 'percent of traffic to redirect to the new branch',
+      required: false,
+    }),
+    end: flags.boolean({
+      description: 'End the rollout.',
+      default: false,
+    }),
+    json: flags.boolean({
+      description:
+        'print output as a JSON object with the new channel ID, name and branch mapping.',
+      default: false,
+    }),
+  };
+
+  async run() {
+    const {
+      args: { channel: channelName },
+      flags: { json: jsonFlag, end: endFlag },
+    } = this.parse(ChannelRollout);
+    let {
+      flags: { branch: branchName, percent },
+    } = this.parse(ChannelRollout);
+
+    const projectDir = await findProjectRootAsync(process.cwd());
+    if (!projectDir) {
+      throw new Error('Please run this command inside a project directory.');
+    }
+    const { exp } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+    const accountName = await getProjectAccountNameAsync(exp);
+    const { slug } = exp;
+    const projectId = await ensureProjectExistsAsync({
+      accountName,
+      projectName: slug,
+    });
+
+    const getUpdateChannelByNameForAppResult = await getUpdateChannelByNameForAppAsync({
+      appId: projectId,
+      channelName: channelName!,
+    });
+    const { branchMapping: currentBranchMapping, isRollout } = getBranchMapping(
+      getUpdateChannelByNameForAppResult
+    );
+
+    if (currentBranchMapping.data.length === 0) {
+      throw new Error('The channel is not pointing at any branches.');
+    }
+    if (currentBranchMapping.data.length > 2) {
+      throw new Error('This CLI cannot handle branch mappings with more than 2 branches.');
+    }
+
+    if (isRollout && branchName && !endFlag) {
```

Comment:
add comment that the user is likely trying to start a rollout here since they have supplied a branch name and no end flag.